examples/platforms/posix/ncp/radio-ncp.c

```diff
@@ -0,0 +1,596 @@
+    IEEE802154_KEY_ID_MODE_2    = 2 << 3,
+    IEEE802154_KEY_ID_MODE_3    = 3 << 3,
+    IEEE802154_KEY_ID_MODE_MASK = 3 << 3,
+
+    IEEE802154_MACCMD_DATA_REQ = 4,
+};
+
+enum
+{
+    kIdle,
+    kSent,
+    kDone,
+};
+
+static uint8_t sReceivePsdu[OT_RADIO_FRAME_MAX_SIZE];
+static uint8_t sTransmitPsdu[OT_RADIO_FRAME_MAX_SIZE];
+static uint8_t sAckPsdu[OT_RADIO_FRAME_MAX_SIZE];
+
+static otRadioFrame sReceiveFrame;
+static otRadioFrame sTransmitFrame;
+static otRadioFrame sAckFrame;
+
+static int      sSockFd = -1;
+static uint8_t  sExtendedAddress[OT_EXT_ADDRESS_SIZE];
+static uint16_t sShortAddress;
+static uint16_t sPanid;
+static uint8_t  sChannel;
+static int8_t   sReceiveSensitivity = 0;
+static bool     sPromiscuous        = false;
+static bool     sAckWait            = false;
+static uint8_t  sTxState            = kIdle;
+
+static otError      sLastTransmitError;
+static otRadioState sState = OT_RADIO_STATE_DISABLED;
+
+static void radioTransmit(otInstance *aInstance);
+const char *dump_hex(const uint8_t *buf, uint16_t len);
+
+static inline otPanId getDstPan(const uint8_t *frame)
+{
+    return (otPanId)((frame[IEEE802154_DSTPAN_OFFSET + 1] << 8) | frame[IEEE802154_DSTPAN_OFFSET]);
+}
+
+static inline otShortAddress getShortAddress(const uint8_t *frame)
+{
+    return (otShortAddress)((frame[IEEE802154_DSTADDR_OFFSET + 1] << 8) | frame[IEEE802154_DSTADDR_OFFSET]);
+}
+
+static inline void getExtAddress(const uint8_t *frame, otExtAddress *address)
+{
+    size_t i;
+
+    for (i = 0; i < sizeof(otExtAddress); i++)
+    {
+        address->m8[i] = frame[IEEE802154_DSTADDR_OFFSET + (sizeof(otExtAddress) - 1 - i)];
+    }
+}
+
+static inline bool isAckRequested(const uint8_t *frame)
+{
+    return (frame[0] & IEEE802154_ACK_REQUEST) != 0;
+}
+
+void otPlatRadioGetIeeeEui64(otInstance *aInstance, uint8_t *aIeeeEui64)
+{
+    otError error = ncpGet(SPINEL_PROP_HWADDR, SPINEL_DATATYPE_EUI64_S, aIeeeEui64);
+    assert(error == OT_ERROR_NONE);
+    OT_UNUSED_VARIABLE(aInstance);
+}
+
+void otPlatRadioSetPanId(otInstance *aInstance, uint16_t panid)
+{
+    otError error = ncpSet(SPINEL_PROP_MAC_15_4_PANID, SPINEL_DATATYPE_UINT16_S, panid);
+    assert(error == OT_ERROR_NONE);
+    sPanid = panid;
+
+    OT_UNUSED_VARIABLE(aInstance);
+}
+
+void otPlatRadioSetExtendedAddress(otInstance *aInstance, const otExtAddress *address)
+{
+    otError error;
+
+    for (size_t i = 0; i < sizeof(sExtendedAddress); i++)
+    {
+        sExtendedAddress[i] = address->m8[sizeof(sExtendedAddress) - 1 - i];
+    }
+
+    error = ncpSet(SPINEL_PROP_MAC_15_4_LADDR, SPINEL_DATATYPE_EUI64_S, sExtendedAddress);
+    // TODO assert should be changed to runtime checks.
+    assert(error == OT_ERROR_NONE);
+
+    OT_UNUSED_VARIABLE(aInstance);
+}
+
+void otPlatRadioSetShortAddress(otInstance *aInstance, uint16_t address)
+{
+    otError error = ncpSet(SPINEL_PROP_MAC_15_4_SADDR, SPINEL_DATATYPE_UINT16_S, address);
+    assert(error == OT_ERROR_NONE);
```

Comment:
Very rarely, but i managed twice to get this assert.

examples/platforms/posix/ncp/radio-ncp.c:175: otPlatRadioSetShortAddress: Assertion `error == OT_ERROR_NONE' failed.

Reply:
Thanks, I will try reproducing this issue.